### PR TITLE
feat(arm): CKV_AZURE_92 to Ensure that Virtual Machines use managed disks

### PR DESCRIPTION
**Pull request by userrut**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sast|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

added new arm policy for resource: VMStorageOsDisk

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my feature, policy, or fix is effective and works
- [X] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/arm/checks/resource/VMStorageOsDisk.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf: Dict[str, Any]) -> CheckResult:
+        properties = conf.get('properties')
+        if not properties:
+            return CheckResult.PASSED
+        storage_profile = properties.get('storageProfile')
+        if not storage_profile:
+            return CheckResult.PASSED
+        os_disk = storage_profile.get('osDisk')
```
Comment (ChanochShayner) [inline checkov/arm/checks/resource/VMStorageOsDisk.py:22]: lets check if `storage_profile` field is a dict before accessing it.

File: checkov/arm/checks/resource/VMStorageOsDisk.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            return CheckResult.PASSED
+        storage_profile = properties.get('storageProfile')
+        if not storage_profile or not isinstance(storage_profile, dict):
+            return CheckResult.PASSED
+        os_disk = storage_profile.get('osDisk')
+        data_disks = storage_profile.get('dataDisks', [])
+        if os_disk and isinstance(os_disk, dict) and "vhd" in os_disk:
+            self.evaluated_keys = ['os_disk']
+            return CheckResult.FAILED
+        if data_disks and any(isinstance(data_disk, dict) and "vhd" in data_disk for data_disk in data_disks):
```
Comment (ChanochShayner) [inline checkov/arm/checks/resource/VMStorageOsDisk.py:27]: data_disk is a list and not a dict

**Review by ChanochShayner — APPROVED** [review]

File: tests/arm/checks/resource/example_VMStorageOsDisk/fail.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            },
+            "vhd": {
+              "uri": "someURI"
+            }
+          }
+        },
+        "osProfile": {
+          "computerName": "hostname",
+          "adminUsername": "testadmin",
+          "adminPassword": "Password1234!"
```
Comment (ChanochShayner) [inline tests/arm/checks/resource/example_VMStorageOsDisk/fail.json:36]:
```suggestion
          "adminPassword": "1111"
```

File: tests/arm/checks/resource/example_VMStorageOsDisk/fail2.json [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+              "vhd": {
+                "uri": "someURI"
+              }
+            }
+          ]
+        },
+        "osProfile": {
+          "computerName": "hostname",
+          "adminUsername": "testadmin",
+          "adminPassword": "Password1234!"
```
Comment (ChanochShayner) [inline tests/arm/checks/resource/example_VMStorageOsDisk/fail2.json:38]:
```suggestion
          "adminPassword": "1111"
```

File: tests/arm/checks/resource/example_VMStorageOsDisk/pass.json [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+            "createOption": "FromImage",
+            "managedDisk": {
+              "storageAccountType": "Standard_LRS"
+            }
+          }
+        },
+        "osProfile": {
+          "computerName": "hostname",
+          "adminUsername": "testadmin",
+          "adminPassword": "Password1234!"
```
Comment (ChanochShayner) [inline tests/arm/checks/resource/example_VMStorageOsDisk/pass.json:33]:
```suggestion
          "adminPassword": "1111"
```

File: tests/arm/checks/resource/example_VMStorageOsDisk/pass2.json [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+            "createOption": "FromImage",
+            "managedDisk": {
+              "storageAccountType": "Standard_LRS"
+            }
+          }
+        },
+        "osProfile": {
+          "computerName": "hostname",
+          "adminUsername": "testadmin",
+          "adminPassword": "Password1234!"
```
Comment (ChanochShayner) [inline tests/arm/checks/resource/example_VMStorageOsDisk/pass2.json:33]:
```suggestion
          "adminPassword": "1111"
```

**Review by achiar99 — APPROVED** [review]

**Pull request merged**